lib/src/utils/config_visitor.dart

```diff
@@ -0,0 +1,78 @@
+import 'package:analyzer/dart/analysis/utilities.dart';
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+
+import 'logging.dart';
+
+/// Visits a `dart_dev/config.dart` file and searches for a line length.
+///
+/// Expects the configuration to be in the format of:
+/// ```dart
+/// import 'package:over_react_format/dart_dev_tool.dart';
+///
+/// final config = {
+///   'format': OverReactFormatTool()
+///     ..lineLength = 120,
+/// };
+/// ```
+class ConfigVisitor extends GeneralizingAstVisitor {
+  int lineLength;
+  bool usesCustomFormatter = false;
+  bool usesOverReactFormat = false;
+
+  ConfigVisitor();
+
+  @override
+  visitMapLiteralEntry(MapLiteralEntry node) {
+    super.visitMapLiteralEntry(node);
+
+    SimpleStringLiteral formatKey = node.key.childEntities.firstWhere((e) {
+      if (e is SimpleStringLiteral) {
+        if (e.value == 'format') {
+          return true;
+        }
+      }
+
+      return false;
+    }, orElse: () => null);
+
+    if (formatKey != null) {
+      if (node.value.runtimeType.toString() != 'FormatTool') {
+        usesCustomFormatter = true;
+
+        if (node.value.runtimeType.toString() != 'OverReactFormatTool') {
```

Comment:
Note to self - this is off and should be (assuming the core logic is good):
```suggestion
        if (node.value.runtimeType.toString() == 'OverReactFormatTool') {
```